### PR TITLE
Chore: convert `LogRowContextProvider` tests to RTL

### LINKED ISSUE #55030: Convert `packages/grafana-ui/src/components/Logs/LogRowContextProvider.test.tsx` to RTL

### PR DESCRIPTION
**Pull request by ashharrison90**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes https://github.com/grafana/grafana/issues/55030

**Special notes for your reviewer**:

### CODE REVIEW

File: packages/grafana-ui/src/components/Logs/LogRowContextProvider.test.tsx [hunk 1]
```diff
@@ -169,7 +169,7 @@ describe('LogRowContextProvider', () => {
         return <></>;
       });
       await act(async () => {
```
Comment (matyax) [inline packages/grafana-ui/src/components/Logs/LogRowContextProvider.test.tsx:171]: We should be able to remove this `act`. If there's an async update, maybe wait for something?

**Review by matyax — COMMENTED** [review]
Left a note.

**Review by gabor — APPROVED** [review]
LGTM!

**Review by matyax — CHANGES_REQUESTED** [review]
Requesting more changes, since I think we can further improve this test.

**Review by matyax — APPROVED** [review]
Awesome!

**Pull request merged**